xwiki-platform-core/xwiki-platform-component/xwiki-platform-component-wiki/src/main/java/org/xwiki/component/wiki/internal/DefaultWikiObjectComponentManagerEventListener.java

```diff
@@ -87,11 +94,13 @@ public void onEvent(Event event, Object source, Object data)
             EntityReference xObject = xObjectEvent.getReference();
 
             // If the modified XObject can produce a WikiComponent
-            if (this.wikiObjectComponentManagerEventListenerProxy.getWikiObjectsList()
-                    .contains(((BaseObjectReference) xObject).getXClassReference())) {
+            WikiObjectComponentBuilder componentBuilder =
+                    this.getAssociatedComponentBuilder(((BaseObjectReference) xObject).getXClassReference());
```

Comment:
I mean that the event give you a XWikiDocument instance (with getOriginalDocument() for delete event) from which you can get the BaseObject and you can then call getXClassReference() on it. And since you have the BaseObject you can also directly pass it to wikiObjectComponentManagerEventListenerProxy.registerObjectComponents